tests/functional/conftest.py

```diff
@@ -180,15 +401,46 @@ def owner(test_accounts):
 
 
 @pytest.fixture
-def contract_type() -> ContractType:
-    return ContractType.parse_obj(RAW_CONTRACT_TYPE)
+def solidity_contract_type() -> ContractType:
+    return ContractType.parse_obj(RAW_SOLIDITY_CONTRACT_TYPE)
+
+
+@pytest.fixture
+def solidity_contract_container(solidity_contract_type) -> ContractContainer:
+    return ContractContainer(contract_type=solidity_contract_type)
+
+
+@pytest.fixture
+def solidity_contract_instance(
+    owner, solidity_contract_container, networks_connected_to_tester
+) -> ContractInstance:
+    return owner.deploy(solidity_contract_container)
 
 
 @pytest.fixture
-def contract_container(contract_type) -> ContractContainer:
-    return ContractContainer(contract_type=contract_type)
+def vyper_contract_type() -> ContractType:
+    return ContractType.parse_obj(RAW_VYPER_CONTRACT_TYPE)
 
 
 @pytest.fixture
-def contract_instance(owner, contract_container) -> ContractInstance:
-    return owner.deploy(contract_container)
+def vyper_contract_container(vyper_contract_type) -> ContractContainer:
+    return ContractContainer(contract_type=vyper_contract_type)
+
+
+@pytest.fixture
+def vyper_contract_instance(
+    owner, vyper_contract_container, networks_connected_to_tester
+) -> ContractInstance:
+    return owner.deploy(vyper_contract_container)
+
+
+@pytest.fixture(params=("solidity", "vyper"))
+def contract_container(
+    request, solidity_contract_container, vyper_contract_container, networks_connected_to_tester
+):
+    return solidity_contract_container if request.param == "solidity" else vyper_contract_container
+
+
+@pytest.fixture(params=("solidity", "vyper"))
```

Comment:
All tests that use this fixture use both a contract compiled with solidity as well as one with vyper and they have almost exactly the same ABI so they work for most tests.